lib/ansible/modules/clustering/k8s/k8s_service.py

```diff
@@ -0,0 +1,618 @@
+            - system.
+          returned: success
+          type: int
+        deletion_timestamp:
+          description:
+            - Number of seconds allowed for this object to gracefully terminate before it will be removed from the
+            - system.
+          returned: success
+          type: string
+    spec:
+      description: The actual configuration of the service
+      returned: success
+      type: complex
+      contains:
+        cluster_ip:
+          description: IP address of the service
+          returned: success
+          type: string
+        external_i_ps:
+          description:
+            - List of IP addresses for which nodes in the cluster will also accept traffic for the service.
+          returned: success
+          type: list
+        external_name:
+          description:
+            - External reference that kubedns or equivalent will return as a CNAME record for this service.
+          returned: success
+          type: string
+        load_balancer_ip:
+          description: If the service is a load balancer, it will get created with the specified IP.
+          returned: success
+          type: string
+        load_balancer_source_ranges:
+          description:
+            - When the service is a load balancer, inbound traffic flow will be restricted to this set of IP
+            - ranges.
+          returned: success
+          type: string
+        ports:
+          description: The list of ports exposed by this service.
+          returned: success
+          type: complex
+        selector:
+          description:
+            - Key, value pairs used to determine which pods to route traffic to.
+          returned: success
+          type: complex
+        session_affinity:
+          description:
+            - Contains either I(ClientIp) or I(None), indicating which form of session affinity is
+            - active, if any.
+          returned: success
+          type: string
+        type:
+          description:
+            - Determines how the Service is exposed. Valid options are I(ExternalName), I(ClusterIP), I(NodePort),
+            - and I(LoadBalancer).
+          returned: success
+          type: string
+'''
+
+
+def create_service(module, kube):
+    '''
+    Create a new service.
+    Assembles a k8s template from module params, and posts it to the API.
+    :param module: Ansible module object
+    :param kube: kubernetes client
+    :param check_mode: bool
+    :return: API response JSON
+    '''
+
+    template = {
+        u'apiVersion': u'v1',
+        u'kind': u'Service',
+        u'metadata': {},
+        u'spec': {}
+    }
+
+    template['metadata']['name'] = module.params['name']
+    if module.params.get('labels'):
+        template['metadata']['labels'] = module.params.get('labels')
+    if module.params.get('annotations'):
+        template['metadata']['annotations'] = module.params.get('annotations')
+    if module.params.get('cluster_ip'):
```

Comment:
I've been thinking that it would be a good idea to prefix parameters with the subcomponent they belong to (with the exception of metadata fields). So anything under spec, could be spec_<field name>